marlin/src/constraints/verifier_key/circuit_verifier_key.rs

```diff
@@ -161,16 +161,17 @@ impl<
         let tmp = value_gen()?;
         let ivk = tmp.borrow();
 
-        let mut index_comms = Vec::<PCG::CommitmentVar>::new();
-        for (i, index_comm) in ivk.circuit_commitments.iter().enumerate() {
-            index_comms.push(PCG::CommitmentVar::alloc(
-                cs.ns(|| format!("index_comm_{}", i)),
-                || Ok(index_comm),
-            )?);
-        }
+        let index_comms: Vec<_> = ivk
+            .circuit_commitments
+            .iter()
+            .enumerate()
+            .map(|(i, index_comm)| PCG::CommitmentVar::alloc(cs.ns(|| format!("index_comm_{}", i)), || Ok(index_comm)))
+            .collect::<Result<_, _>>()?;
 
         // `alloc_constant` regardless of the mode.
-        let verifier_key = PCG::VerifierKeyVar::alloc_constant(cs.ns(|| "verifier_key"), || Ok(&ivk.verifier_key))?;
+        // TODO(pratyush): investigate if this can be alloc_constant or not.
+        // (I think it cannot).
```

Comment:
It's not okay to allocate as a constant, but in `PCG::VerifierKeyVar::alloc` I allocate the appropriate fields as constants.